service_crategen/src/commands/generate/codegen/rest_json.rs

```diff
@@ -113,11 +113,15 @@ impl GenerateProtocol for RestJsonGenerator {
         Ok(())
     }
 
-    fn generate_prelude(&self, writer: &mut FileWriter, _: &Service) -> IoResult {
-        writeln!(writer,
-                 "use serde_json;
-        use rusoto_core::param::{{Params, ServiceParams}};
-        use rusoto_core::signature::SignedRequest;
+    fn generate_prelude(&self, writer: &mut FileWriter, service: &Service) -> IoResult {
+        writeln!(writer, "use serde_json;")?;
+
+        // avoid unused imports when building services that don't use params
+        if service.service_type_name() != "Batch" {
```

Comment:
Not a huge fan of having service customizations being done in the generator. Could this be resolved by just putting a `#![allow(unused_imports)]` in the module?